File: problems/Python/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dmap = {}
        
        for s in strs:
            key = "".join(sorted(s))
            if key not in dmap:
                dmap[key] = []
            dmap[key].append(s)
            
        return list(dmap.values())

File: problems/C++/49. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;
        
        for (const string& s : strs) {
            string key = s;
            sort(key.begin(), key.end());
            map[key].push_back(s);
        }
        
        vector<vector<string>> res;
        for (auto const& pair : map) {
            res.push_back(pair.second);
        }
        
        return res;
    }
};